src/wallet/wallet.cpp

```diff
@@ -2537,6 +2523,11 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             return false; // TODO: Allow non-wallet inputs
     }
 
+    if (coin_control.HasSelected() && !coin_control.fAllowOtherInputs)
+    {
+        return (nValueRet >= nTargetValue);
```

Comment:
What about `setCoinsRet`?

Reply:
`nValueRet` is never updated, so this will always return false.

Reply:
So it will always fail when `coin_control.fAllowOtherInputs` is not set?